### PR TITLE
Support using the liquidatable accounts feed as data source

### PR DESCRIPTION
**Pull request by ckamm**
This adds a new `LIQUIDATABLE_FEED_WEBSOCKET_ADDRESS` environment variable. If set, the liquidator will try to liquidate only accounts passed in through this feed.

Documentation is still missing.

### CODE REVIEW

File: src/liquidator.ts [hunk 1]
```diff
@@ -291,6 +265,98 @@ async function main() {
...
+    }
+  } catch (err) {
+    console.error('Error liquidating account:', err);
+  }
+}
+
+// never returns
+async function liquidatableFromLiquidatableFeed() {
+  const ws = new RpcWebSocketClient(liquidatableFeedWebsocketAddress, {
+    max_reconnects: Infinity,
```
Comment (microwavedcola1) [inline src/liquidator.ts:347]: what happens when your remove websocket server dies, should you switch to the default mode?

File: src/liquidator.ts [hunk 2]
```diff
@@ -343,17 +344,26 @@ async function newAccountOnLiquidatableFeed(params) {
...
+      if (!candidatesSet.has(account)) {
+        candidatesSet.add(account);
+        candidates.enqueue(account);
+      }
+  });
 
-  // fully reactive now
   while (true) {
-    await sleep(5000);
+    const account = await candidates.dequeue();
```
Comment (microwavedcola1) [inline src/liquidator.ts:364]: what if the queue is empty? then it returns an empty promise? but then that is not a mango account, or?

Comment (ckamm) [reply src/liquidator.ts:364]: If the queue is empty this blocks until a new element is added, then returns it. (it returns a promise that will only be resolved by the next `enqueue()`)

File: src/liquidator.ts (hunk 1 above)
Comment (ckamm) [reply src/liquidator.ts:350]: I assume the websocket server will restart and then we'll want to reconnect to it. I don't want to build fallover behavior.

**Comment by ckamm** [comment]
@riordanp The liquidation test passes for me now. Would be happy to get a review!

**Review by riordanp — APPROVED** [review]
Looks good to me, wondered about some failover behaviour but it could get pretty complex and I don't think it's totally necessary. Maybe some notification on websocket disconnect/close would be helpful

**Pull request merged**